example-get-started-experiments/code/sagemaker/deploy_model.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import re
+import sys
+
+from sagemaker.deserializers import JSONDeserializer
+from sagemaker.pytorch import PyTorchModel
+from sagemaker.serverless import ServerlessInferenceConfig
+
+
+memory_size = { 
+    "dev": 4096 ,
+    "staging": 4096,
+    "prod": 6144 ,
+    "default": 4096,
+}
+max_concurrency = { 
+    "dev": 5,
+    "staging": 5,
+    "prod": 10,
+    "default": 5,
+}
+
+
+def deploy(
+    name: str,
+    stage: str,
+    version: str,
+    model_data: str,
+    role: str,
+):
+    sagemaker_logger = logging.getLogger("sagemaker")
+    sagemaker_logger.setLevel(logging.DEBUG)
+    sagemaker_logger.addHandler(logging.StreamHandler(sys.stdout))
+
+    composed_name =  re.sub(
+        r"[^a-zA-Z0-9\-]", "-", f"{name}-{version}-{stage}")
+    
+    model = PyTorchModel(
```

Comment:
@daavoo If I created a SageMaker model for a given model version before deploying it in one environment (eg, `dev`), does this code recreate the same model before deploying it in another environment (eg, `prod`). Or does it skip recreating it and just return a reference to the old model?

Reply:
It currently creates a new model. If we want to do what you said last, we should only add `stage` to the composed name after the model has been created.

Reply:
Referring to [this msg](https://iterativeai.slack.com/archives/C01SR9Q12LB/p1690971158412739?thread_ts=1690817210.212859&cid=C01SR9Q12LB), you said that the existing model would be re-used, right? Would removing `stage` from the model name make this happen?

Reply:
> Would removing stage from the model name make this happen?

Yes

Reply:
ok, I'll create a PR to remove stage from the model name.

Reply:
[PR](https://github.com/iterative/example-repos-dev/pull/236)